docs: Move Join operator to public preview (#277)
axiomhq/docs: apl/tabular-operators/join-operator.mdx

- Enrich existing data with supplementary details.

<Note>
    The `join` operator is currently in public preview. For more information, see [Feature states](/getting-started-guide/feature-states).

>     The preview of the `join` operator works with variable limits depending on the structure of your dataset. For the left side of the join, the limit is 50,000 rows when the dataset has fewer than 100 fields. This limit decreases linearly as the field count increases: 25,000 rows at 200 fields, 12,500 rows at 400 fields, and 10,000 rows at 500+ fields. The right side of the join has a consistent limit of 50,000 rows.

Reviewer: ```suggestion
    The preview of the `join` operator works with variable limits depending on the structure of your dataset. For the left side of the join, the limit decreases linearly as the field count increases:
        - 50,000 rows for datasets with fewer than 100 fields
        - 25,000 rows for datasets with up to 200 fields
        - 12,500 rows for datasets with up to 400 fields
        - 10,000 rows for datasets with 500 fields or more
    The right side of the join has a consistent limit of 50,000 rows.
```

Reviewer: Thanks for finding out these details. I have converted it to a list for better readability but a table might work even better.

I was wondering about the cut-off points. The original text said `the limit is 50,000 rows when the dataset has fewer than 100 fields ... 25,000 rows at 200 fields, 12,500 rows at 400 fields, and 10,000 rows at 500+ fields.` Am I right to interpret this that the limit changes to 25,000 rows at 200 fields? If yes, the limit is 50,000 rows for datasets with up to 200 rows, right?
My other interpretation is that the limit is 25,000 rows for datasets with 101 to 200 fields. But then I'm not sure what happens between 400 and 500. Should I find this out or do you already know the answer?

Reviewer 2: It is a smooth linear interpolation, so it doesn't jump in steps. Ie, if you pick some random number like 287 fields, then it'll be adjusted to that.

A table with some other numbers:
```
numFields vs dynamicMaxLimit
1 50000
34 50000
67 50000
100 50000
133 37593
166 30120
199 25125
232 21551
265 18867
298 16778
331 15105
364 13736
397 12594
430 11627
463 10799
496 10080
```

Reviewer: Thanks a lot! I don't know if users expect this level of specificity, but I'm happy to put this in the docs if this is the case. What do you think @dominicchapman ?

Reviewer 2: you text also says that it decreases linearly, so I think it is already correct

Reviewer: Ah, I get it now. The original text only gave some examples. In this case, it's fine, I would just add a short thing: https://github.com/axiomhq/docs/pull/277/commits/48eb716a22d26a836269ac3e8b4ef45b90394df3
